mirdata/datasets/saraga_hindustani.py

```diff
@@ -0,0 +1,497 @@
+                'start_time': float(start_time)
+                if '.' in start_time
+                else int(start_time),
+                'duration': float(duration) if '.' in duration else int(duration),
+            }
+
+            section_count += 1  # Go to next section
+
+    return tempo_annotation
+
+
+def load_sama(sama_path):
+    """Load sama
+
+    Args:
+        sama_path (str): Local path where the sama annotation is stored.
+            If `None`, returns None.
+
+    Returns:
+        SectionData: sama annotations
+
+    """
+    if sama_path is None:
+        return None
+
+    if not os.path.exists(sama_path):
+        raise IOError("sama_path {} does not exist".format(sama_path))
+
+    timestamps = []
+    sama_cycles = []
+    intervals = []
+    with open(sama_path, 'r') as reader:
+        for line in reader.readlines():
+            timestamps.append(float(line))
+
+    for i in np.arange(1, len(timestamps)):
+        intervals.append([timestamps[i - 1], timestamps[i]])
+        sama_cycles.append('sama cycle ' + str(i))
+
+    if not intervals:
+        return None
+
+    return utils.SectionData(np.array(intervals), sama_cycles)
+
+
+def load_sections(sections_path):
+    """Load tracks sections
+
+    Args:
+        sections_path (str): Local path where the section annotation is stored.
+
+    Returns:
+        SectionData: section annotations for track
+
+    """
+    if sections_path is None:
+        return None
+
+    if not os.path.exists(sections_path):
+        raise IOError("sections_path {} does not exist".format(sections_path))
+
+    intervals = []
+    section_labels = []
+
+    with open(sections_path, 'r') as reader:
+        for line in reader.readlines():
+            if line != '\n':
+                intervals.append(
+                    [
+                        float(line.split(',')[0]),
+                        float(line.split(',')[0]) + float(line.split(',')[2]),
+                    ]
+                )
+                section_labels.append(
+                    str(line.split(',')[3].split('\n')[0])
+                    + '-'
+                    + str(line.split(',')[1])
+                )
+
+    if not intervals:
+        return None
+
+    return utils.SectionData(np.array(intervals), section_labels)
+
+
+def load_phrases(phrases_path):
+    """Load phrases
+
+    Args:
+        phrases_path (str): Local path where the phrase annotation is stored.
+            If `None`, returns None.
+
+    Returns:
+        EventData: phrases annotation for track
+
+    """
+    if phrases_path is None:
+        return None
+
+    if not os.path.exists(phrases_path):
+        raise IOError("phrases_path {} does not exist".format(phrases_path))
+
+    start_times = []
+    end_times = []
+    events = []
+    with open(phrases_path, 'r') as reader:
+        for line in reader.readlines():
+            if len(line.split('\t')) == 4:
+                start_times.append(float(line.split('\t')[0]))
```

Comment:
it looks like 
```
start_times.append(float(line.split('\t')[0]))
                end_times.append(
                    float(line.split('\t')[0]) + float(line.split('\t')[2])
                )
```
is common code to both if branches and what it is different is the events

Reply:
I don't understand what you mean here @nkundiushuti. For both Carnatic and Hindustani collections, melodic phrase annotations are the same. 
First column: start time, second column: flag to identify how representative is the phrase, third column: duration, and fourth column: solfège symbols to identify the phrases.

The length condition here is because when testing the full dataset in the first Saraga version, I found that some melodic phrase annotation rows had no fourth column, or in other words, no annotations of the solfège symbols. **That issue is not happening for the whole annotation file but just in some rows**, that is why I set this condition here, to prevent the `load_phrases` function from crashing, as well as noticing the users that the phrase annotation in certain timestamp is not available.

Reply:
Sorry @genisplaja , I didn't explain it well. What I meant is to replace this: 
```
with open(phrases_path, 'r') as reader:
        for line in reader.readlines():
            if len(line.split('\t')) == 4:
                start_times.append(float(line.split('\t')[0]))
                end_times.append(
                    float(line.split('\t')[0]) + float(line.split('\t')[2])
                )
                events.append(str(line.split('\t')[3].split('\n')[0]))
            if len(line.split('\t')) == 3:
                start_times.append(float(line.split('\t')[0]))
                end_times.append(
                    float(line.split('\t')[0]) + float(line.split('\t')[2])
                )
                events.append('No information')
```
with this: 
```
with open(phrases_path, 'r') as reader:
        for line in reader.readlines():
           start_times.append(float(line.split('\t')[0]))
                end_times.append(
                    float(line.split('\t')[0]) + float(line.split('\t')[2])
                )
            if len(line.split('\t')) == 4:
                events.append(str(line.split('\t')[3].split('\n')[0]))
            else:
                events.append('No information')
```

Reply:
Ok, I see! Now I understand. Thanks!